documentation/specs/Notification.md

```diff
@@ -0,0 +1,567 @@
+# `Notification` Component Specification
+
+## Overview
+
+The `Notification` component is capable of displaying `alerts` and `toasts` messages through a `useNotification` hook. Use of this component requires that your app consume `EasyUIProvider`.
+
+Toasts dispay a brief non-disruptive message to the user as a result of an action taken. The displayed message is accompanied by an associated status icon and cannot be manually dismissed, instead toast messages automatically timeout after `4000ms`. Toasts should not be used to display critical information such as a system failure, use `alerts` for such behavior. They should also contain purely text and no elements such as links.
+
+Alerts display an important message near the top of the page that does not automatically timeout and requires user action to dismiss. The displayed messaged is accompanied by an associated status icon. If the message is not prominent enough to immediately require the user's attention, consider using toasts.
+
+### Toast Use Cases
+
+- Inform user of a completed action.
+- Inform user that an asynchronous process has been triggered based on their action.
+
+### Alert Use Cases
+
+- Inform user of system level errors such as a failed network request.
+- Inform user of any critical information that requires their attention.
+
+### Features
+
+- Notifications are interacted with through the use of a `useNotification` hook.
+- Toasts have ARIA `role="status"` and alerts have ARIA `role="alert"`.
+- The `message` prop is of type `ReactNode`.
+
+### Risks and Challenges
+
+- Unlike other components, `Notification` components are unique in that they may be quite opinionated with regards to how they display and the kind of control they expose to consumers.
+- Building for the shipper's experience at EasyPost according to internal specs while maintaining a pleasant DX for outside consumers.
+- Allowing consumers to customize placement.
+
+### Prior Art
+
+- [Paste `<Alert />`](https://paste.twilio.design/components/alert)
+- [Paste `<Toast />`](https://paste.twilio.design/components/toast)
+- [Spectrum `<Toast />`](https://react-spectrum.adobe.com/react-spectrum/Toast.html)
+
+---
+
+## Design
+
+### API
+
+```typescript
+import type { AriaToastProps } from "@react-aria/toast";
+import type { QueuedToast } from "@react-stately/toast";
+
+export type NotificationType = "alert" | "toast";
+
+export type NotificationStatus =
+  | "promotional"
+  | "success"
+  | "neutral"
+  | "error"
+  | "warning";
```

Comment:
i hesitated a little to use these names since `neutral` is a property on the `Button` that means something entirely different, but based on this architecture, consumers shouldn't really have to worry about passing these props directly and instead they'll use `showNeutralToast` and `showNeutralAlert`, which i think is a bit better, but open to ideas.